templates/spec/system/search_spec.rb

```diff
@@ -39,6 +39,15 @@
       expect(page.all('[data-search-target="result"]')[0][:class]).to include('autocomplete-results__item--current')
     end
 
+    it 'allows mouse click on results' do
+      fill_in 'keywords', with: 'ruby'
+      # wait for the autocomplete to show up
+      expect(page.all('[data-search-target="result"]')[0][:class]).to include('autocomplete-results__item--current')
```

Comment:
Are you sure that the `include` RSpec matcher will [wait](https://github.com/teamcapybara/capybara#asynchronous-javascript-ajax-and-friends) until the item is present?

Reply:
I think it's more `page.all` that is waiting here, from the [`#all` doc](https://rubydoc.info/github/jnicklas/capybara/master/Capybara%2FNode%2FFinders:all):

> By default Capybara's waiting behavior will wait up to [default_max_wait_time](https://rubydoc.info/github/jnicklas/capybara/master/Capybara#configure-class_method) seconds for matching elements to be available and then return an empty result if none are available.

So, you are right that this is not ideal, meaning that we would probably have the same effect without the `include('autocomplete-results__item--current')` part. 

Because this is used in multiple tests over this file, if that's ok for you I'd merge this and shoot another PR that cleans this up in the whole file.

Reply:
Makes sense 👍